### PR TITLE
 [user-provided-ci-tags] add DD_CI metadata

### PR DESCRIPTION
**Pull request by KevinYeramian**
[card](https://datadoghq.atlassian.net/browse/SW-1773)
### What and why?

- add `DD_CI_XXX` metadata to allow user to override them with env variables.

### How?

- Add `getUserCIMetadata` function and use it in `getCIMetadata`

### CODE REVIEW

File: src/helpers/__tests__/ci.test.ts [hunk 1]
```diff
@@ -45,13 +89,51 @@ describe('getCIMetadata', () => {
...
+      DD_GIT_COMMIT_COMMITTER_EMAIL: 'DD_GIT_COMMIT_COMMITTER_EMAIL',
+      DD_GIT_COMMIT_COMMITTER_NAME: 'DD_GIT_COMMIT_COMMITTER_NAME',
+      DD_GIT_COMMIT_MESSAGE: 'DD_GIT_COMMIT_MESSAGE',
+      DD_GIT_COMMIT_SHA: 'DD_GIT_COMMIT_SHA',
+      DD_GIT_REPOSITORY_URL: 'DD_GIT_REPOSITORY_URL',
+      DD_GIT_TAG: 'DD_GIT_TAG',
+    }
+
+    const expectedMetadata = ciAppTagsToMetadata(ddMetadataToSpanTags(DD_METADATA))
+    delete expectedMetadata.git.branch
```
Comment (juan-fernandez) [inline src/helpers/__tests__/ci.test.ts:118]: why delete the branch?

File: src/helpers/__tests__/user-provided-git.test.ts [hunk 2]
```diff
@@ -71,18 +85,57 @@ describe('getUserGitMetadata', () => {
...
-    delete process.env.DD_GIT_COMMIT_COMMITTER_EMAIL
-    delete process.env.DD_GIT_COMMIT_COMMITTER_DATE
-
+    process.env = {}
     const result = getUserGitMetadata()
     expect(result).toEqual({})
   })
 })
+
+describe('getUserGitMetadata', () => {
```
Comment (juan-fernandez) [inline src/helpers/__tests__/user-provided-git.test.ts:94]: is this a typo?
```suggestion
describe('getUserCIMetadata', () => {
```

File: src/helpers/ci.ts [hunk 3]
```diff
@@ -486,8 +486,10 @@ export const getCISpanTags = (): SpanTags | undefined => {
 export const getCIMetadata = (): Metadata | undefined => {
   const tags = {
     ...getCISpanTags(),
+    ...getUserCIMetadata(),
     ...getUserGitMetadata(),
```
Comment (juan-fernandez) [inline src/helpers/ci.ts:490]: there's a bit of confusion with `get...Metadata` and `get...SpanTags`. Sometimes `get...Metadata` returns a plain dictionary string to string (like `getUserCIMetadata` and `getUserGitMetadata`) and sometimes it returns a "complex" dictionary, like `getCIMetadata`. I'm not saying it should be fixed in this PR, just pointing it out. Maybe renaming every function that returns a plain string:string dictionary to `get...SpanTags` might be a good idea.

**Review by juan-fernandez — APPROVED** [review]

**Comment by juan-fernandez** [comment]
could we rename the title of the PR from `user-provided-git` to `user-provided-ci-tags` or `user-provided-metadata` ? It's not quite `user-provided-git`, which we've used before and can lead to confusion

File: src/helpers/ci.ts (hunk 3 above)
Comment (KevinYeramian) [reply src/helpers/ci.ts:490]: I completely agree, when I tried to add  `getUserCIMetadata` it was already confusing.

File: src/helpers/__tests__/ci.test.ts (hunk 1 above)
Comment (KevinYeramian) [reply src/helpers/__tests__/ci.test.ts:118]: `git.branch` is the only edge case and can break depending of tests cases

File: src/helpers/ci.ts (hunk 3 above)
Comment (KevinYeramian) [reply src/helpers/ci.ts:490]: https://github.com/DataDog/datadog-ci/pull/460

**Review by m-rousse — APPROVED** [review]
